addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-cidr-service.js

```diff
@@ -113,26 +122,62 @@ class EnvironmentScCidrService extends Service {
           'The Security Group for this workspace does not contain any ingress rules configured in the Service Catalog product template',
           true,
         );
-
       // Perform the actual security group ingress rule updates
       const newCidrList = await this.getUpdatedIngressRules(requestContext, {
         existingEnvironment,
         securityGroupId,
         currentIngressRules,
         updateRequest,
       });
-
       // Not storing the changes in the DB, but since all went well
       // we return the cidr field as part of the env obj
       existingEnvironment.cidr = newCidrList;
     });
-
+    // As we removed the 443 Port CIDRs values for Rstudio during the ingress rule updates.
+    // So once it is done we replace the original values to the existing env.
+    if (productName === 'RStudioV2') {
+      cloneUpdateRequest = JSON.parse(cloneUpdateRequest);
+      existingEnvironment.cidr = cloneUpdateRequest;
+    }
     // Write audit event
     await this.audit(requestContext, { action: 'update-environment-sc-cidr', body: existingEnvironment });
-
     return existingEnvironment;
   }
 
+  async modifyELBRule(existingEnvironment, updateRequest, albService, requestContext) {
```

Comment:
Same comment that I made about modifyRule function

Reply:
modifyELBRule function declaration as included into the existing locking mechanism. Please review